app/uk/gov/hmrc/trusts/services/AuthService.scala

```diff
@@ -29,16 +30,16 @@ import scala.util.control.NonFatal
 @Singleton
 class AuthService @Inject()(override val authConnector: AuthConnector) extends AuthorisedFunctions {
 
-   def authorisedUser()(f: => Future[Result])
-                     (implicit hc: HeaderCarrier): Future[Result] = {
-    authorised() {
-      f
+
+  def authorisedUser()(f: Option[AffinityGroup] => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
+    authorised().retrieve(affinityGroup) {
```

Comment:
Can we do `authorised().retrieve(affinityGroup).map(f)` here?

Reply:
Done @adamconder